### PR TITLE
[MBL-1721] Pledge Rewards Summary Amount Alignment

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

- aligns the amount labels with the title label instead of the header label to match designs.
- removes redundant "Add-Ons" headers from summary cells
- updates "Bonus support" cell text to ksr_black

# 🤔 Why

designs are the source of truth

# 🛠 How

adjust the stack view alignment and constrain the amount label to the title lable

# 👀 See

| Before 🐛 | After 🦋 |
| --- | --- |
| Simulator Screenshot - iPhone 15 Pro 17 5 - 2024-09-11 at 10 31 13 | Simulator Screenshot - iPhone 15 Pro 17 5 - 2024-09-11 at 10 08 57 |

# ✅ Acceptance criteria

- [x] summary table matches designs

### CODE REVIEW

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/Features/PledgePaymentMethods/Views/Cells/PostCampaignPledgeRewardsSummaryCell.swift:97:1 | warning: (trailingSpace) Remove trailing space at end of a line. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by ifosli — APPROVED** [review]
Looks good! There are so many screenshots here, and a lot of them feel very similar. It'd be nice if we could delete/reduce some of the tests to make changes easier to review? (Ie either just do one language/screen size per scenario, or drop some of the scenarios if they're not changing the ui.) That does not need to happen in this pr, though!

File: Kickstarter-iOS/Features/PledgePaymentMethods/Views/Cells/PostCampaignPledgeRewardsSummaryCell.swift [hunk 1]
```diff
@@ -119,7 +119,7 @@ private func headerLabelStyle(_ label: UILabel) {
 
 private func labelStyle(_ label: UILabel) {
   label.font = UIFont.ksr_subhead().bolded
-  label.textColor = UIColor.ksr_support_400
+  label.textColor = label.text == Strings.Bonus_support() ? UIColor.ksr_black : UIColor.ksr_support_400
```
Comment (ifosli) [inline Kickstarter-iOS/Features/PledgePaymentMethods/Views/Cells/PostCampaignPledgeRewardsSummaryCell.swift:122]: Nice catch! I did double check this in the figma, and there's a bunch of other little things we could be doing in this table, including adding a per-item cost when there's multiple of an add-on, which I literally just noticed was in there.

File: Kickstarter-iOS/Features/PledgeView/Controllers/__Snapshots__/NoShippingPledgeViewControllerTests/testView_PledgeContext_FixPaymentMethod_ErroredCard.lang_en_device_phone4_7inch.png
Comment (ifosli) [inline Kickstarter-iOS/Features/PledgeView/Controllers/__Snapshots__/NoShippingPledgeViewControllerTests/testView_PledgeContext_FixPaymentMethod_ErroredCard.lang_en_device_phone4_7inch.png:1]: Oh no! We never fixed the wording for the "pledge without a reward" case! I'll file a ticket.

**Review by ifosli — APPROVED** [review]

**Pull request merged**